File: 36-valid-sudoku/valid-sudoku.go
import "strconv"

func isValidSudoku(board [][]byte) bool {
    set := map[string]bool{};

    for i := 0 ; i < 9 ; i++{
        for j := 0 ; j < 9 ; j++ {

            if board[i][j] == '.' {continue}



            row  := "row" + strconv.Itoa(i) + "-" + string(board[i][j]);
            col  := "col" + strconv.Itoa(j) + "-" + string(board[i][j]);
            grid := "grid" + strconv.Itoa(i/3) + "-" + strconv.Itoa(j/3) + "-" + string(board[i][j]);

            if set[row] || set[col] || set[grid] {
                return false;
            }

            set[row] = true;
            set[col] = true;
            set[grid] = true;

        }
    }

    return true;
}

File: 36-valid-sudoku/valid-sudoku.py
class Solution:
    def isValidSudoku(self, board: List[List[str]]) -> bool:
        s = set()

        for i in range(9):
            for j in range(9):

                if board[i][j] == "." : continue

                row = f"row,{i},{board[i][j]}"
                col = f"col,{j},{board[i][j]}"
                grid = f"grid,{i//3},{j//3},{board[i][j]}"

                if row in s or col in s or grid in s:
                    return False

                s.add(row)
                s.add(col)
                s.add(grid)
        
        return True